Xbim.IO.MemoryModel/Xml/XbimXmlReader4.cs

```diff
@@ -830,10 +857,18 @@ private string GetTextFromElement(XmlReader input)
             return input.Value;
         }
 
-        public static IStepFileHeader ReadHeader(Stream input)
+        [Obsolete]
+        public static IStepFileHeader ReadHeader(Stream input, ILogger logger = null)
+        {
+            var xReader = new XbimXmlReader4(default, logger);
+            var fakeModel = new Memory.MemoryModel(new Ifc4.EntityFactoryIfc4(), logger);
+            return xReader.Read(input, fakeModel, true); //using a dummy model to get the assembly correct
+        }
+
+        public static IStepFileHeader ReadHeader(Stream input, ILoggerFactory loggerFactory)
```

Comment:
We should look to remove all loggers from methods, and just leave to Constructors (with ServiceLocator as a fallback?)

Reply:
Issue is: these are mostly on static 'factory' methods. I'd propose we drop the ILoggerFacrory param on static factory methods, and just resort to a Service Locator when needed. Longer term we should replace the static methods with real factories

Reply:
Removed them all. All (?) added earlier in this PR anyway